android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/CreateEngineSessionMiddleware.kt

```diff
@@ -35,7 +35,8 @@ internal class CreateEngineSessionMiddleware(
         action: BrowserAction,
     ) {
         if (action is EngineAction.CreateEngineSessionAction) {
-            if (context.state.findTabOrCustomTab(action.tabId)?.engineState?.initializing == false) {
+            val engineState = context.state.findTabOrCustomTab(action.tabId)?.engineState
+            if (engineState?.initializing == false && (engineState.engineSession == null || engineState.crashed)) {
```

Comment:
I believe we will want this to be `!engineState.crashed`, otherwise we will just hit the early return [here](https://searchfox.org/mozilla-mobile/rev/c942971ddfdf1b73078e0a552f423f3abcc7524f/firefox-android/android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/CreateEngineSessionMiddleware.kt#96). `crashed` is reset during [restoration of a crashed tab](https://searchfox.org/mozilla-mobile/rev/c942971ddfdf1b73078e0a552f423f3abcc7524f/firefox-android/android-components/components/browser/state/src/main/java/mozilla/components/browser/state/reducer/CrashReducer.kt#27)

Reply:
In fact, we probably want `engineState?.initializing == false && engineState.engineSession == null && !engineState.crashed`